### PR TITLE
Properly handle `exp_horizon` + zkid.move VK & Configuration initialization in Rust

### PR DESCRIPTION
**Pull request by alinush**
### Description

Fixes some things in #11895.

### CODE REVIEW

File: aptos-move/vm-genesis/src/lib.rs [hunk 1]
```diff
@@ -525,6 +531,31 @@ fn initialize_on_chain_governance(session: &mut SessionExt, genesis_config: &Gen
     );
 }
 
+fn initialize_zkid(session: &mut SessionExt) {
```
Comment (heliuchuan) [inline aptos-move/vm-genesis/src/lib.rs:534]: should we perhaps check the chain id to be non-main?

theoretically in a dooms day scenario if we restarted main, would it load the devnet vk?

File: aptos-move/aptos-vm/src/zkid_validation.rs [hunk 2]
```diff
@@ -132,6 +141,10 @@ pub fn validate_zkid_authenticators(
         match &zkid_sig.sig {
             ZkpOrOpenIdSig::Groth16Zkp(proof) => match jwk {
                 JWK::RSA(rsa_jwk) => {
+                    if proof.exp_horizon_secs > config.max_exp_horizon_secs {
```
Comment (heliuchuan) [inline aptos-move/aptos-vm/src/zkid_validation.rs:144]: can you add a todo for me to write a smoke test for this later

**Review by heliuchuan — APPROVED** [review]

**Review by zekun000 — APPROVED** [review]

File: aptos-move/vm-genesis/src/lib.rs (hunk 1 above)
Comment (alinush) [reply aptos-move/vm-genesis/src/lib.rs:534]: I think you mean a mainnet wipe, not restart. We will never wipe mainnet or testnet though.

File: aptos-move/aptos-vm/src/zkid_validation.rs (hunk 2 above)
Comment (alinush) [reply aptos-move/aptos-vm/src/zkid_validation.rs:144]: I think there is one in the smoke tests

File: types/src/zkid.rs [hunk 3]
```diff
@@ -292,11 +311,18 @@ pub struct Groth16Zkp {
...
+    /// A signature on the proof (via the training wheels SK) to mitigate against flaws in our circuit
     pub training_wheels_signature: EphemeralSignature,
+    /// An extra field (e.g., `"<name>":"<val>") that will be matched publicly in the JWT
     pub extra_field: String,
+    /// Will be set to the override `aud` value that the circuit should match, instead of the `aud` in the IDC.
+    /// This will allow users to recover their zkID accounts derived by an application that is no longer online.
     pub override_aud_val: Option<String>,
+    /// The expiration horizon that the circuit should enforce on the expiration date committed in the nonce.
+    /// This must be <= `Configuration::max_expiration_horizon_secs`.
+    pub exp_horizon_secs: u64,
```
Comment (alinush) [inline types/src/zkid.rs:325]: cc @larry-aptos

**Pull request merged**